### DISCUSSION TITLE
Creating full "page" components

### DISCUSSION THREAD

**Discussion by samturrell** [nuxt/nuxt#20858] [Questions]
Hi there!

In nuxt 2, we basically had a setup where each page was a component itself, and the pages directory would just be used for routing and export a single component, for example we would have something like:

`components/pages/HomePage/HomePage.vue` <-- All the components and logic for the homepage
`pages/index.vue` <-- A route page that would export the `HomePage.vue` component

`pages/index.vue` would contain something like the following:

```
<script>
import HomePage from '../components/pages/HomePage/HomePage.vue';`

export default {
    name: 'home-page',

    extends: HomePage,
};
</script>
```

This would allow the `index.vue` route to render the full component defined by the `HomePage.vue` file.

With this format we could feed our full page components into our storybook stories, visual regression testing and unit testing.

With Nuxt 3 this gets a bit more complicated with the introduction of macros, notably `definePageMeta` which needs to be included at the page level and not in a component for it to work.

Is there any way for this to still be achievable in nuxt 3?

## What i've tried:

Change `index.vue` to `index.js` and just export the `HomePage.vue`
Change the `index.vue` `<script setup>` to a `<script>` and export the `HomePage.vue` as the default export
Have `index.vue` just render the `HomePage.vue` component
Update the `index.vue` to use the options api with `extends: HomePage,`

**Comment by maou-shonen** [comment]
I haven't tried it, but what do you think about this?

```vue
// ~/components/pages/HomePage/HomePage.vue
<script>
import Page from '~/pages/Index.vue'
export default Page
</script>
```

**Reply by samturrell** [reply]
Hmm... Yeah i imagine this would work. Bit of a reversed approach, but could be the only way to do it

**Reply by samturrell** [reply]
It seems like with this approach HMR doesn't seem to work when making changes within the page but viewing the component

**Reply by samturrell** [reply]
Weirdly, making changes in a nested component causes that component to HMR correctly, but the parent component doesn't update
